### PR TITLE
feat: add story video preview

### PR DESCRIPTION
**Pull request by ice-alcides**
Figma [link](https://www.figma.com/design/3TnZngoFeklOVC8TexlR1D/%F0%9F%A7%8A-ICE-Cloud?node-id=40427-127876&node-type=frame&t=HWVxkO4RGECmiVtZ-0)

- A preview of the recorded video for Story

Limitations in these changes that are going to be resolved in other pull requests: 

- A share button shows a settings dialog only without closing a story preview page. I added a todo-comment to improve it for the next PR
- Currently, StoryPreview shows the videos, but we also need to support a preview of the images for this flow.

CleanShot 2024-10-15 at 14 22 09@2x

### CODE REVIEW

File: lib/app/features/feed/create_story/views/pages/story_preview_page.dart [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    super.key,
+  });
+
+  final String videoPath;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return Scaffold(
+      body: SafeArea(
+        child: Column(
```
[the remark this answers is not in the document]
Comment (ice-alcides) [reply lib/app/features/feed/create_story/views/pages/story_preview_page.dart:26]: good question, I will update

File: lib/app/features/feed/create_story/views/components/story_video/story_video_preview.dart [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+                final videoAspectRatio = videoController.value.aspectRatio;
+
+                return SizedBox(
+                  width: constraints.maxWidth,
+                  height: constraints.maxWidth / videoAspectRatio,
+                  child: VideoPlayer(videoController),
+                );
+              },
+              orElse: () => const Center(child: IceLoadingIndicator()),
```
Comment (ice-tychon) [inline lib/app/features/feed/create_story/views/components/story_video/story_video_preview.dart:51]: You already have `LoadingIndicator` which is the same.

Comment (ice-kreios) [inline lib/app/features/feed/create_story/views/components/story_video/story_video_preview.dart:51]: I see that you've already created the LoadingIndicator component, which renders a centered IceLoadingIndicator. Could we use it for this line instead?

File: lib/app/features/feed/create_story/views/components/story_video/loading_indicator.dart [hunk 3]
```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:flutter/material.dart';
+import 'package:ice/app/components/progress_bar/ice_loading_indicator.dart';
+
+class LoadingIndicator extends StatelessWidget {
```
Comment (ice-tychon) [inline lib/app/features/feed/create_story/views/components/story_video/loading_indicator.dart:6]: This class isn't used anywhere

File: lib/app/features/feed/create_story/views/pages/story_preview_page.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (ice-alcides) [reply lib/app/features/feed/create_story/views/pages/story_preview_page.dart:26]: I refactored this part a little, the preview widget will be automatically adjusted to the screen size

File: lib/app/features/feed/create_story/views/components/story_video/loading_indicator.dart (hunk 3 above)
Comment (ice-alcides) [reply lib/app/features/feed/create_story/views/components/story_video/loading_indicator.dart]: changed

File: lib/app/features/feed/create_story/views/components/story_video/story_video_preview.dart (hunk 2 above)
Comment (ice-alcides) [reply lib/app/features/feed/create_story/views/components/story_video/story_video_preview.dart]: updated

Comment (ice-alcides) [reply lib/app/features/feed/create_story/views/components/story_video/story_video_preview.dart]: done

File: lib/app/features/feed/create_story/hooks/use_camera_recording_controller.dart [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    if (isCameraReady.value) {
+      await ref.read(storyCameraControllerProvider.notifier).startVideoRecording();
+    }
+  }
+
+  Future<void> handleRecordingStop() async {
+    if (isCameraReady.value) {
+      final videoPath = await ref.read(storyCameraControllerProvider.notifier).stopVideoRecording();
+      if (videoPath != null && context.mounted) {
+        await StoryPreviewRoute(videoPath: videoPath).push<void>(context);
```
Comment (ice-kreios) [inline lib/app/features/feed/create_story/hooks/use_camera_recording_controller.dart:54]: Could we handle the route on the UI side to keep this function more flexible and reusable? This way, we could adapt it for different screens without needing to modify this hook.

Comment (ice-alcides) [reply lib/app/features/feed/create_story/hooks/use_camera_recording_controller.dart]: done

**Review by ice-kreios — APPROVED** [review]

**Pull request merged**